controllers/numaresourcesoperator_controller.go

```diff
@@ -449,17 +449,16 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 	klog.V(4).InfoS("RTESync start", "trees", len(trees))
 	defer klog.V(4).Info("RTESync stop")
 
-	errorList := dangling.DeleteUnusedDaemonSets(r.Client, ctx, instance, trees)
-	if len(errorList) > 0 {
-		klog.ErrorS(fmt.Errorf("failed to delete unused daemonsets"), "errors", errorList)
+	err := dangling.DeleteUnusedDaemonSets(r.Client, ctx, instance, trees)
+	if err != nil {
+		klog.ErrorS(fmt.Errorf("failed to delete unused daemonsets"), "errors", err)
```

Comment:
this needs to be rewritten:
`klog.ErrorS(err, "failed to deleted unused daemonsets")`

can be done in a later PR

Reply:
updated. Thanks for your valuable review rounds